### PR TITLE
Feat/1787 site map redo

### LINKED ISSUE #1787: [Dev-ready Design]
**Description:**
Provide grantees an alternative route to access TDP links in the form of a site map. 

**Related User Stories:**

—As a [Grantee] I want a site map page so that I can easily access TDP links to help me get to where I need to go next. 

**Recommended pa11y actions:**

—‘click element #error-report’,—

**AC:**

- [ ] Documentation work for the following has occurred:
   - [ ] Relevant User stories.
   - [ ] Recommended pa11y checks.
   - [ ] Updating living UX documents, e.g. User Flows or Personas(if relevant).
- [ ] Dev/Design sync has occurred; resulting tickets created
- [ ] The design is usable and accessible, meaning it adheres to definition of done standards for design work.
> <details>
>      <summary>Expand Definition of Done Standards</summary>
> 
> - It uses [USWDS components and follows it’s UX guidance](https://designsystem.digital.gov/components/), or a deviation is clearly documented
> 
> - Language is intentional and [plain](https://plainlanguage.gov/guidelines/); placeholders are clearly documented
> - It follows [accessibility guidelines](https://accessibility.digital.gov/) and accessibility implementation notes are documented (e.g. clear information hierarchy, color is not the only way meaning is communicated, etc.)
> - If feedback identifies bigger questions or unknowns, create additional issues to investigate
> </details>

**Tasks:**

- [ ] —Draft design in Figma—

**Supporting Documentation:**

—Link to Figma and Dev-Ready Design folder PDF—

### PR DESCRIPTION
**Pull request by n0remac**
## Summary of Changes
_Provide a brief summary of changes_
Pull request closes #1787 

## How to Test
_List the steps to test the PR_
These steps are generic, please adjust as necessary.
```
cd tdrs-frontend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d
cd tdrs-backend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d 
```

1. Open http://localhost:3000/, on the spash page there will be no sitemap
2. Sign in and the sitemap link will appear at the bottom.
3. Go to site map and test the links.
4. Accounts with admin access will see the admin page link in the site map.
5. Accounts without admin access will not.

Screen Shot 2022-08-11 at 9 55 10 AM
Screen Shot 2022-08-11 at 9 55 17 AM
Screen Shot 2022-08-11 at 9 55 29 AM

### CODE REVIEW

**Comment by n0remac** [comment]
I changed to a new branch because I rebased the old branch from the wrong spot. This is a fresh branch from develop with all the relevant commits cherry-picked from one of the old branches.

This branch resolves the link target and data file issues that @reitermb noted in the last pr.

**Comment by reitermb** [comment]
> I changed to a new branch because I rebased the old branch and messed up the history. This is a fresh branch from develop with all the relevant commits cherry-picked from one of the old branches.
> 
> This branch resolves the link target and data file issues that @reitermb noted in the last pr.

Link target is working great now, still seeing the data files access issue though. Screen recorded the flow here in case that's helpful

https://user-images.githubusercontent.com/28515874/185298591-c6381c7c-6323-4403-9647-07fb34a014f2.mp4

File: tdrs-frontend/src/utils/canViewAdmin.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const isMemberOfOne = (user, groupNames) =>
+  user?.roles?.some((role) => groupNames.includes(role.name))
+
+const userAccessRequestApproved = (user) =>
+  user?.['access_request'] && user?.roles?.length > 0
+
+const canViewAdmin = (user) =>
+  userAccessRequestApproved(user) &&
+  isMemberOfOne(user, ['Developer', 'OFA System Admin', 'ACF OCIO'])
```
Comment (ADPennington) [inline tdrs-frontend/src/utils/canViewAdmin.js:9]: @n0remac `OFA Admin` also has access to backend, but the admin tab hasnt been visible for this group for some reason. a user with this role can submit data on behalf of any STT.  if its easy to include the tab and group in this, that'd be great. else, we can spin up a new ticket.

**Review by andrew-jameson — APPROVED** [review]

**Comment by ADPennington** [comment]
@n0remac please checkout [this comment](https://github.com/raft-tech/TANF-app/pull/2069#discussion_r949088374)

@ttran-hub this is ready for `gov a11y review`. Please note:
- please use the [qasp environment](https://tdp-frontend-qasp.app.cloud.gov/). Note that you will need to use a non-acf email address and have `ofa system` role to access the full site map. home, data files, profile, and site-map pages need a11y review. Ping me with any questions 😄 
- raft's a11y review is [here](https://github.com/raft-tech/TANF-app/pull/2069#issuecomment-1219043876).

**Comment by ttran-hub** [comment]
> @n0remac please checkout [this comment](https://github.com/raft-tech/TANF-app/pull/2069#discussion_r949088374)
> 
> @ttran-hub this is ready for `gov a11y review`. Please note:
> 
> * please use the [qasp environment](https://tdp-frontend-qasp.app.cloud.gov/). Note that you will need to use a non-acf email address and have `ofa system` role to access the full site map. home, data files, profile, and site-map pages need a11y review. Ping me with any questions 😄
> * raft's a11y review is [here](https://github.com/raft-tech/TANF-app/pull/2069#issuecomment-1219043876).

@reitermb @n0remac The Broken ARIA reference error was flagged in the H1 header for the Site Map, Home, Data Files, and Profile pages. Please check, otherwise, the pages look good. cc @ADPennington

**Comment by reitermb** [comment]
> > @n0remac please checkout [this comment](https://github.com/raft-tech/TANF-app/pull/2069#discussion_r949088374)
> > @ttran-hub this is ready for `gov a11y review`. Please note:
> > 
> > * please use the [qasp environment](https://tdp-frontend-qasp.app.cloud.gov/). Note that you will need to use a non-acf email address and have `ofa system` role to access the full site map. home, data files, profile, and site-map pages need a11y review. Ping me with any questions 😄
> > * raft's a11y review is [here](https://github.com/raft-tech/TANF-app/pull/2069#issuecomment-1219043876).
> 
> @reitermb @n0remac The Broken ARIA reference error was flagged in the H1 header for the Site Map, Home, Data Files, and Profile pages. Please check, otherwise, the pages look good. cc @ADPennington

We should be able to disregard that—it's there to trigger a read of alert banners after the h1's, so when they don't exist it'll pop a please review error but shouldn't adversely impact anything! Note: If you go to the Profile page while its in a pending Request Access state you shouldn't see that error since the banner element exists there.

**Comment by ADPennington** [comment]
> > > @n0remac please checkout [this comment](https://github.com/raft-tech/TANF-app/pull/2069#discussion_r949088374)
> > > @ttran-hub this is ready for `gov a11y review`. Please note:
> > > 
> > > * please use the [qasp environment](https://tdp-frontend-qasp.app.cloud.gov/). Note that you will need to use a non-acf email address and have `ofa system` role to access the full site map. home, data files, profile, and site-map pages need a11y review. Ping me with any questions 😄
> > > * raft's a11y review is [here](https://github.com/raft-tech/TANF-app/pull/2069#issuecomment-1219043876).
> > 
> > 
> > @reitermb @n0remac The Broken ARIA reference error was flagged in the H1 header for the Site Map, Home, Data Files, and Profile pages. Please check, otherwise, the pages look good. cc @ADPennington
> 
> We should be able to disregard that—it's there to trigger a read of alert banners after the h1's, so when they don't exist it'll pop a please review error but shouldn't adversely impact anything! Note: If you go to the Profile page while its in a pending Request Access state you shouldn't see that error since the banner element exists there.

thank you @reitermb @ttran-hub -- if possible today, I'd like to pair for 15 min to better understand this behavior before signing off.

**Comment by ADPennington** [comment]
> > > > @n0remac please checkout [this comment](https://github.com/raft-tech/TANF-app/pull/2069#discussion_r949088374)
> > > > @ttran-hub this is ready for `gov a11y review`. Please note:
> > > > 
> > > > * please use the [qasp environment](https://tdp-frontend-qasp.app.cloud.gov/). Note that you will need to use a non-acf email address and have `ofa system` role to access the full site map. home, data files, profile, and site-map pages need a11y review. Ping me with any questions 😄
> > > > * raft's a11y review is [here](https://github.com/raft-tech/TANF-app/pull/2069#issuecomment-1219043876).
> > > 
> > > 
> > > @reitermb @n0remac The Broken ARIA reference error was flagged in the H1 header for the Site Map, Home, Data Files, and Profile pages. Please check, otherwise, the pages look good. cc @ADPennington
> > 
> > 
> > We should be able to disregard that—it's there to trigger a read of alert banners after the h1's, so when they don't exist it'll pop a please review error but shouldn't adversely impact anything! Note: If you go to the Profile page while its in a pending Request Access state you shouldn't see that error since the banner element exists there.
> 
> thank you @reitermb @ttran-hub -- if possible today, I'd like to pair for 15 min to better understand this behavior before signing off.

@reitermb and I met on this and he was able to show me that this error is flagged for review given the missing alert banner as he mentions here. Also confirmed that this has no impact on the user. Because there is no alert to read to the user, this will not be read to them. cc: @ttran-hub

**Review by ADPennington — APPROVED** [review]
looks good. thanks @n0remac please include the qasp checklist in the summary of changes going forward. Including it here for reference:

---
## Deliverables
_More details on how deliverables herein are assessed included [here](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
- [x] As a Grantee, I want a site map page so that I can easily access TDP links to help me get to where I need to go next
- [x] Add Site Map link to footer in the Authenticated scenario
- [x] Testing Checklist has been run and all tests pass
+ [x] **`lfrohlich`** and/or **`adpennington`**  confirmed that ACs are met.

### [Deliverable 2: Tested Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [x] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [x] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [x] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [x] Are backend code style checks passing on CircleCI?
+ [x] Are frontend code style checks passing on CircleCI?
+ [x] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [x] Does this PR complete the epic? 
+ [x Does PR include documentation for Raft's a11y review? 
+ [x] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?
_yes but upon review, determined it was okay_

### [Deliverable 5: Deployed](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [x] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [ ] Does this PR provide background for why coding decisions were made?
+ [x] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] Can reviewer explain and take ownership of these elements presented in this code review?

_more documentation on changes would be helpful to include in the comments of file changes. for future devs. much of this was discussed in tabletop_


### [Deliverable 7: Secure](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [x] Does the OWASP Scan pass on CircleCI?

### [Deliverable 8: User Research](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

_n/a_

**Comment by ttran-hub** [comment]
> 

Thanks @reitermb and @ADPennington, this pull request has already been merged.  Therefore, it is not urgent and if your schedule permits, can we sync up to catchup on the ARIA error for future references and review. Thanks

**Pull request merged**